iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ValidationUtil.java

```diff
@@ -383,13 +518,302 @@ public static void checkPhaseTapChangerRegulation(Validable validable, PhaseTapC
+            throw new ValidationException(validable, REGULATING_TERMINAL_NOT_DEFINED);
+        }
+        if (!validVoltageSetpoint(voltageSetpoint)) {
+            throw createInvalidValueException(validable, voltageSetpoint, VOLTAGE_SETPOINT);
+        }
+    }
+
+    private static void checkReactivePowerTarget(Validable validable, double targetQ) {
+        if (Double.isNaN(targetQ)) {
+            throw createInvalidValueException(validable, targetQ, REACTIVE_POWER_SETPOINT, VOLTAGE_REGULATOR_OFF);
+        }
+    }
+
+    /**
+     * @deprecated
+     * Use {@link #checkVoltageSetpoint(Validable, String, double, boolean)} instead
+     */
+    @Deprecated(since = "4.4.0")
+    public static boolean checkVoltageControl(Validable validable, Boolean voltageRegulatorOn, double voltageSetpoint) {
+        if (voltageRegulatorOn == null) {
+            throw new ValidationException(validable, "voltage regulator status is not set");
+        }
+        boolean bVoltageRegulatorOn = voltageRegulatorOn; // make sonar happy java:S5411 Boxed "Boolean" should be avoided in boolean expressions
+        if (bVoltageRegulatorOn) {
+            if (Double.isNaN(voltageSetpoint) || voltageSetpoint <= 0) {
+                throw createInvalidValueException(validable, voltageSetpoint, VOLTAGE_SETPOINT, "voltage regulator is on");
+            }
+            return false;
+        }
+        return true;
+    }
+
+    public static boolean validRegulatingVoltageControl(Terminal regulatingTerminal, double voltageSetpoint, Network network) {
+        return validRegulatingTerminal(regulatingTerminal, network) && validVoltageSetpoint(voltageSetpoint);
+    }
+
+    public static boolean validRegulatingVoltageControl(Terminal regulatingTerminal, double voltageSetpoint, double targetDeadband, Network network) {
+        return validRegulatingTerminal(regulatingTerminal, network) && validVoltageSetpoint(voltageSetpoint) && validDeadband(targetDeadband);
+    }
+
+    public static boolean validRegulatingVoltageControl(double voltageSetpoint) {
+        return validVoltageSetpoint(voltageSetpoint);
+    }
+
+    public static boolean validRegulatingReactivePowerControl(Terminal regulatingTerminal, double reactivePowerSetpoint, Network network) {
+        return validRegulatingTerminal(regulatingTerminal, network) && validReactivePowerSetpoint(reactivePowerSetpoint);
+    }
+
+    public static boolean validRegulatingCurrentOrActivePowerControl(Terminal regulatingTerminal, double valueSetpoint, double targetDeadband, Network network) {
+        return validRegulatingTerminal(regulatingTerminal, network) && validCurrentOrActivePowerSetpoint(valueSetpoint) && validDeadband(targetDeadband);
+    }
+
+    private static boolean validRegulatingTerminal(Terminal regulatingTerminal, Network network) {
+        return regulatingTerminal != null && regulatingTerminal.getVoltageLevel().getNetwork() == network;
```

Comment:
It seems that it never checks that regulatingTerminal != null if regulating?

Reply:
I do not quite understand what you mean. Could you please clarify this point?

Reply:
After looking at the code, it seems it is a mistake on my part, please discard this comment.